src/server/service/diagnostics.rs

```diff
@@ -1484,26 +1484,25 @@ mod log {
             req.set_levels(vec![LogLevel::Warn.into()].into());
             req.set_patterns(vec![".*test-filter.*".to_string()].into());
             let expected = vec![
-                "2019/08/23 18:11:58.387 +08:00",
                 "2019/08/23 18:09:58.387 +08:00",
+                "2019/08/23 18:11:58.387 +08:00",
```

Comment:
Why are they out of order?

Reply:
Found that the order was messed because of `log_file3` had a record with inconsistent timestamp with the rest it the file.

Reply:
Then why the test case passes before?

Reply:
The expected results was ordered by timestamp desc then, which is not right. And due to `log_file` and `log_file3` having the same timestamps at the first log record, the searching order was `log_file` -> `log_files3`. 
Actually, it confused me that this seems an unstable test when the `search_files` are already sorted by `file_start_time` (the ts of first log record in the file). Still figuring this out.

Reply:
After re-visiting the code, the unstable order of search files might be caused by `read_dir` and `tempdir` on different platforms. But I fail to re-capture this on my own desktop :( .

Reply:
Seems the search result need to be sorted in this test to make it stable?

Reply:
As the target search logs have ordered timestamps at the first record now (they were the same before), the sorting here should be expected and stable by this:
https://github.com/tikv/tikv/blob/3e460b9c2669eea733cefdcce027a09926cc4e44/src/server/service/diagnostics.rs#L911-L915